typed-protocols-examples/src/Network/TypedProtocol/ReqResp/Codec.hs

```diff
@@ -19,15 +21,20 @@ codecReqResp
 codecReqResp =
     Codec{encode, decode}
   where
-    encode :: forall (pr :: PeerRole) st st'.
-              PeerHasAgency pr st
+    encode :: forall (req :: *) (resp :: *)
```

Comment:
Is the kind `*` really needed here?

Reply:
@coot, not really, turns out (it was, when I was on the intermediate steps of fixing the compilation).  Fixed.